File: server/models/order.go
// 订单模型（映射 orders 表）
package models

import "time"

// Order 订单表结构。
// 订单状态机（见 schema 注释）：
//
//	0待支付 1待取(已付租金) 2租赁中(押金冻结) 3待归还 4已归还(结算) 5已取消 6违约(扣押金)
type Order struct {
	Id           int64     `orm:"column(id);auto" json:"id"`
	OrderNo      string    `orm:"column(order_no);size(64);unique" json:"order_no"` // 订单号
	ItemId       int64     `orm:"column(item_id)" json:"item_id"`
	RenterId     int64     `orm:"column(renter_id)" json:"renter_id"` // 租客用户ID
	OwnerId      int64     `orm:"column(owner_id)" json:"owner_id"`   // 房东用户ID
	StartDate    string    `orm:"column(start_date);type(date)" json:"start_date"`
	EndDate      string    `orm:"column(end_date);type(date)" json:"end_date"`
	Days         int       `orm:"column(days)" json:"days"` // 租赁天数
	RentAmount   float64   `orm:"column(rent_amount);digits(12);decimals(2)" json:"rent_amount"`
	Deposit      float64   `orm:"column(deposit);digits(12);decimals(2)" json:"deposit"`
	Status       int       `orm:"column(status)" json:"status"` // 状态机见注释
	PayTradeNo   string    `orm:"column(pay_trade_no);size(64)" json:"pay_trade_no"`
	CancelReason string    `orm:"column(cancel_reason);size(255)" json:"cancel_reason"`
	CreatedAt    time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
	UpdatedAt    time.Time `orm:"column(updated_at);auto_now;type(datetime)" json:"updated_at"`
}

// TableName 指定表名
func (o *Order) TableName() string {
	return "orders"
}

// 订单状态常量（与 schema 注释对应）
const (
	OrderStatusPending   = 0 // 待支付
	OrderStatusToPickup  = 1 // 待取（已付租金）
	OrderStatusRenting   = 2 // 租赁中（押金冻结）
	OrderStatusToReturn  = 3 // 待归还
	OrderStatusReturned  = 4 // 已归还（结算）
	OrderStatusCancelled = 5 // 已取消
	OrderStatusBreach    = 6 // 违约（扣押金）
)

File: apps/flutter/lib/models/order.dart
import 'user.dart';

/// Order 模型,字段对齐 server/models/order.go。
class Order {
  final int id;
  final String orderNo;
  final int itemId;
  final int renterId;
  final int ownerId;
  final UserProfile? owner; // 详情接口富化:房东公开信息(列表不含)
  final UserProfile? renter; // 详情接口富化:租客公开信息(列表不含)
  final String startDate;
  final String endDate;
  final int days;
  final double rentAmount; // 元
  final double deposit; // 元
  final int status;
  final String payTradeNo;
  final String cancelReason;
  final String createdAt;

  Order({
    required this.id,
    required this.orderNo,
    required this.itemId,
    required this.renterId,
    required this.ownerId,
    this.owner,
    this.renter,
    required this.startDate,
    required this.endDate,
    required this.days,
    required this.rentAmount,
    required this.deposit,
    required this.status,
    required this.payTradeNo,
    required this.cancelReason,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> j) => Order(
        id: j['id'] as int? ?? 0,
        orderNo: j['order_no'] as String? ?? '',
        itemId: j['item_id'] as int? ?? 0,
        renterId: j['renter_id'] as int? ?? 0,
        ownerId: j['owner_id'] as int? ?? 0,
        owner: j['owner'] is Map<String, dynamic>
            ? UserProfile.fromJson(j['owner'] as Map<String, dynamic>)
            : null,
        renter: j['renter'] is Map<String, dynamic>
            ? UserProfile.fromJson(j['renter'] as Map<String, dynamic>)
            : null,
        startDate: j['start_date'] as String? ?? '',
        endDate: j['end_date'] as String? ?? '',
        days: j['days'] as int? ?? 0,
        rentAmount: (j['rent_amount'] as num? ?? 0).toDouble(),
        deposit: (j['deposit'] as num? ?? 0).toDouble(),
        status: j['status'] as int? ?? 0,
        payTradeNo: j['pay_trade_no'] as String? ?? '',
        cancelReason: j['cancel_reason'] as String? ?? '',
        createdAt: j['created_at'] as String? ?? '',
      );

  String get statusText => switch (status) {
        0 => '待支付',
        1 => '待取货',
        2 => '租赁中',
        3 => '待归还',
        4 => '已归还',
        5 => '已取消',
        6 => '违约',
        _ => '未知($status)',
      };
}

List<Order> parseOrderList(Map<String, dynamic> d) =>
    (d['orders'] as List<dynamic>? ?? [])
        .whereType<Map<String, dynamic>>()
        .map(Order.fromJson)
        .toList();
